core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/filter/FilterPredicate.java

```diff
@@ -20,6 +23,11 @@
     @JsonProperty(value = "condition", required = true)
     public ComparisonType condition;
 
+    @JsonSchemaInject(strings = {
+            @JsonSchemaString(path = HideAnnotation.hideTarget, value = "condition"),
```

Comment:
I did not know this `@JsonSchemaString` annotation. My understanding is that it will construct a JSON string based on the values. If so can we update other custom annotations that we use for JsonSchemaInject to do the same? right now we are writing pure JSON string I believe.

Reply:
So unfortunately, the @JsonSchemaString has the limitation that it only merges a flat schema (in that you can't nest strings together in a map). Because of this, some of the annotations (e.g. UIWidget raw json) requires nesting are better done via JSON. However, I'll definitely migrate the flat ones.